pkg/bpf/detect.go

```diff
@@ -79,3 +83,31 @@ func HasKprobeMulti() bool {
 	kprobeMulti.initialized = true
 	return kprobeMulti.detected
 }
+
+func detectBuildId() bool {
+	attr := &unix.PerfEventAttr{
+		Type:        unix.PERF_TYPE_SOFTWARE,
+		Config:      unix.PERF_COUNT_SW_BPF_OUTPUT,
+		Bits:        unix.PerfBitWatermark | unix.PerfBitMmap | unix.PerfBitMmap2 | PerfBitBuildId,
+		Sample_type: unix.PERF_SAMPLE_RAW,
+		Wakeup:      1,
+	}
+
+	attr.Size = uint32(unsafe.Sizeof(*attr))
+	fd, err := unix.PerfEventOpen(attr, -1, 0, -1, unix.PERF_FLAG_FD_CLOEXEC)
+	if err == nil {
+		unix.Close(fd)
+		return true
+	}
+	return false
+}
+
+func HasBuildId() bool {
+	if buildid.initialized {
+		return buildid.detected
+	}
```

Comment:
Should we use sync.Once here to be on the safe side?

Reply:
good idea.. I converted the rest of the features as well, thanks